### PR TITLE
Python CAN Parsing 

### PR DESCRIPTION
**Pull request by Akashem06**

### CODE REVIEW

**Review by mitchellostler — APPROVED** [review]
This all looks good to me. I think I already provided feedback in our meeting so you can go ahead and merge this if it's all good to go. 

As your testing things, and you think of any improvements you want to make to the tool, feel free to add them on! 

I'm excited about this, good work

File: can/tools/can.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
+import struct
+import subprocess
+import time 
+import threading
+
+SYSTEM_CAN_MESSAGE_BABYDRIVER_BABYDRIVER = 2016
```
Comment (JarvisWeng) [inline can/tools/can.py:6]: Hm you could probably autogen a `.py` file and then just import that instead

File: libraries/codegen/templates/can.py.jinja [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{% set boards = data["Boards"] -%}
+{% set messages = data["Messages"] -%}
+
+import struct
+import subprocess
+import time 
+import threading
+
+{% for message in messages -%}
```
Comment (JarvisWeng) [inline libraries/codegen/templates/can.py.jinja:9]: Oh if `can.py` is already an autogenerated file then don't commit it

**Review by JarvisWeng — APPROVED** [review]

File: can/tools/can.py (hunk 1 above)
Comment (Bafran) [reply can/tools/can.py:6]: If this file is autogenerated, add it to the .gitignore and delete this file so it's not in the PR

**Pull request merged**